plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -2386,12 +2386,41 @@ read_only::get_table_rows_result read_only::get_kv_table_rows_context( const rea
    return result;
 }
 
+struct table_receiver
+  : chain::backing_store::table_only_error_receiver<table_receiver, chain::contract_table_query_exception> {
+   table_receiver(read_only::get_table_by_scope_result& result, const read_only::get_table_by_scope_params& params)
+   : result_(result), params_(params) {
+      check_limit();
+   }
+
+   void add_table_row(const backing_store::table_id_object_view& row) {
+      if( params_.table && row.table != params_.table ) {
+         return;
+      }
+      result_.rows.push_back( {row.code, row.scope, row.table, row.payer, row.count} );
+      check_limit();
+   }
+
+   auto keep_processing_entries() {
+      kp_ = keep_processing{};
+      return [&kp=*kp_,&reached_limit=reached_limit_]() {
+         return !reached_limit && kp();
+      };
```

Comment:
If `keep_processing_entries` is called more than once then the second call will destroy the referenced kp in the returned captured lambda of the previous call.

Reply:
Any hints on how to fix this one? I tried several ways to initialize kp but always got compiler errors. Thanks.

Reply:
No reason to have a `std::optional<keep_processing>` as part of the class. Just have `keep_prcoessing_entries` create a `keep_processing` that is moved into the lambda.

Reply:
Thanks Kevin!

Reply:
Fixed.